### PR TITLE
Remove container as public property on Symfony.

### LINKED ISSUE #2954: [BC] Container should not be a variable on Symfony2 module
Container variable should be refactored into a method call to $this->kernel->getContainer() to avoid trouble dealing with non-booted kernels and with tests with multiple requests, as each request boots its own kernel and creates a new container instance.

### PR DESCRIPTION
**Pull request by raistlin**
Fixes #2954

### CODE REVIEW

File: src/Codeception/Module/Symfony.php [hunk 1]
```diff
@@ -350,6 +346,14 @@ public function grabServiceFromContainer($service)
     }
 
     /**
+     * @return \Symfony\Component\DependencyInjection\ContainerInterface
+     */
+    public function getContainer()
```
Comment (DavertMik) [inline src/Codeception/Module/Symfony.php]: should be `_getContainer` for not to appear in functional tests

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,7 @@
 * [Phalcon] Merged `Phalcon1` and `Phalcon2` modules into one `Phalcon` due the fact that Phalcon Framework v1.3.x no longer supported at all
 * *Breaking* [Dbh] module removed
 * *Breaking* [Laravel4] module removed
+* *Breaking* [Symfony] Removed public property `container` you can use new `getContainer` method
```
Comment (DavertMik) [inline CHANGELOG.md]: use new `_getContainer` method

File: src/Codeception/Module/Symfony.php [hunk 3]
```diff
@@ -350,6 +346,14 @@ public function grabServiceFromContainer($service)
     }
 
     /**
+     * @return \Symfony\Component\DependencyInjection\ContainerInterface
+     */
+    public function _getContainer()
+    {
+        return $this->kernel->getcontainer();
```
Comment (DavertMik) [inline src/Codeception/Module/Symfony.php]: getContainer

**Comment by raistlin** [comment]
Agh! CHANGELOG fixed

File: src/Codeception/Lib/Connector/Symfony.php [hunk 4]
```diff
@@ -10,11 +10,26 @@ class Symfony extends \Symfony\Component\HttpKernel\Client
...
      */
     public $persistentServices = [];
 
     /**
+     * Symfony constructor.
+     * @param \Codeception\Module\Symfony $module
+     */
+    public function __construct($module)
+    {
+        $this->module = $module;
```
Comment (DavertMik) [inline src/Codeception/Lib/Connector/Symfony.php]: I'd rather not have cyclic references here. Is that important for connector to be aware of module?

Comment (raistlin) [inline src/Codeception/Lib/Connector/Symfony.php]: If connector must create new kernels, connector must be able to set new kernel on symfony module.
It's not mandatory for this branch, but it's a step forward to allow this behaviour.

**Comment by raistlin** [comment]
- Travis, Facebook failed.
- AppVeyor, I don't have any experience but seems that download on win64openssl failed ...

**Comment by raistlin** [comment]
Potentially invalidated with #2968

**Pull request closed without merging**